packages/grafana-ui/src/components/SelectNG/AsyncSelectNG.tsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import { SelectableValue } from '@grafana/data';
+import { Fetch } from './Fetch';
+import Downshift from 'downshift';
+import { AsyncSelectNGProps } from './types';
+import { SelectMenu, SelectMenuMessage } from './SelectMenu';
+import {
+  DEFAULT_ERROR_MESSAGE,
+  DEFAULT_LOADING_MESSAGE,
+  DEFAULT_NO_OPTIONS_MESSAGE,
+  DEFAULT_PLACEHOLDER,
+  itemToString,
+  renderClearButton,
+  renderDefaultTrigger,
+  renderDropdownButton,
+} from './utils';
+import { useAsyncSelect } from './hooks';
+
+export function AsyncSelectNG<T>({
+  onChange,
+  loadOptions,
+  clearable,
+  disabled,
+  filterable,
+  placeholder = DEFAULT_PLACEHOLDER,
+  noOptionsMessage = DEFAULT_NO_OPTIONS_MESSAGE,
+  loadingMessage = DEFAULT_LOADING_MESSAGE,
+  errorMessage = DEFAULT_ERROR_MESSAGE,
+  placement = 'auto-start',
+}: AsyncSelectNGProps<T>) {
+  const {
+    triggerRef,
+    popperProps,
+    currentValue,
+    onLoadOptions,
+    setPopperRef,
+    setPopperElement,
+    setCurrentValue,
+  } = useAsyncSelect<T>({
+    placement,
+    loadOptions,
+  });
+  return (
+    <>
+      <Downshift
+        itemToString={itemToString}
+        onSelect={onChange}
+        onInputValueChange={v => {
+          setCurrentValue(v);
+        }}
+      >
+        {({
+          getInputProps,
+          getToggleButtonProps,
+          getMenuProps,
+          getItemProps,
+          isOpen,
+          highlightedIndex,
+          selectedItem,
+          clearSelection,
+          closeMenu,
+          openMenu,
+        }) => {
+          const dropdownButton = renderDropdownButton(triggerRef, isOpen, !!disabled, getToggleButtonProps);
+          const clearButton = clearable && selectedItem && renderClearButton(!!disabled, clearSelection);
+          const triggerElement = renderDefaultTrigger(
+            triggerRef,
+            placeholder,
+            isOpen,
+            !!disabled,
+            !!clearable,
+            !!filterable,
+            dropdownButton,
+            clearButton,
+            getInputProps,
+            {
+              openMenu,
+              closeMenu,
+            }
+          );
```

Comment:
Signatures of these functions need to be more aligned

Reply:
The same comment about abstracting this to a separate component.